masq_lib/src/multi_config.rs

```diff
@@ -80,20 +80,39 @@ impl<'a> MultiConfig<'a> {
         Ok(MultiConfig { arg_matches })
     }
 
-    pub fn make_configurator_error(e: clap::Error) -> ConfiguratorError {
-        let invalid_value_regex =
-            Regex::new("Invalid value for.*'--(.*?) <.*? (.*)$").expect("Bad regex");
-        if let Some(captures) = invalid_value_regex.captures(&e.message) {
-            let name = &captures[1];
-            let message = format!("Invalid value: {}", &captures[2]);
-            return ConfiguratorError::required(name, &message);
+    fn check_for_inavlid_value_err(
+        error_msg: &str,
+        pattern: &str,
+        index_of_name: usize,
+        index_of_value: usize,
+    ) -> Option<ConfiguratorError> {
+        let regex = Regex::new(pattern).expect("Bad regex");
+
+        match regex.captures(error_msg) {
+            Some(captures) => {
+                let name = &captures[index_of_name];
+                let message = format!("Invalid value: {}", &captures[index_of_value]);
+
+                Some(ConfiguratorError::required(name, &message))
+            }
+            None => None,
         }
-        let another_invalid_value_regex =
-            Regex::new("error: (.*) isn't a valid value for '--(.*?) <.*>'").expect("Bad regex"); // TODO: is this good enough?
-        if let Some(captures) = another_invalid_value_regex.captures(&e.message) {
-            let name = &captures[2];
-            let message = format!("Invalid value: {}", &captures[1]);
-            return ConfiguratorError::required(name, &message);
+    }
+
+    pub fn make_configurator_error(e: clap::Error) -> ConfiguratorError {
+        let invalid_value_patterns = vec![
+            ("Invalid value for.*'--(.*?) <.*? (.*)$", 1, 2),
```

Comment:
This regex could be improved, I think. First, the `.*?` should probably just be `.*`. Second, if there was a closing `>`, it would be easier to read, even though it's not necessary.